includes/abilities-api/class-wp-abilities-category-registry.php

```diff
@@ -0,0 +1,240 @@
+		if ( ! doing_action( 'abilities_api_categories_init' ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				sprintf(
+					/* translators: 1: abilities_api_categories_init, 2: category slug. */
+					esc_html__( 'Categories must be registered during the %1$s action. The category %2$s was not registered.' ),
+					'<code>abilities_api_categories_init</code>',
+					'<code>' . esc_html( $slug ) . '</code>'
+				),
+				'0.3.0'
+			);
+			return null;
+		}
+
+		if ( $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				esc_html( sprintf( __( 'Category "%s" is already registered.' ), $slug ) ),
+				'0.3.0'
+			);
+			return null;
+		}
+
+		if ( ! preg_match( '/^[a-z0-9]+(?:-[a-z0-9]+)*$/', $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'Category slug must contain only lowercase alphanumeric characters and dashes.' ),
+				'0.3.0'
+			);
+			return null;
+		}
+
+		/**
+		 * Filters the category arguments before they are validated and used to instantiate the category.
+		 *
+		 * @since 0.3.0
+		 *
+		 * @param array<string,mixed> $args The arguments used to instantiate the category.
+		 * @param string              $slug The slug of the category.
+		 */
+		$args = apply_filters( 'register_ability_category_args', $args, $slug );
+
+		try {
+			// WP_Ability_Category::prepare_properties() will throw an exception if the properties are invalid.
+			$category = new WP_Ability_Category( $slug, $args );
+		} catch ( \InvalidArgumentException $e ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html( $e->getMessage() ),
+				'0.3.0'
+			);
+			return null;
+		}
+
+		$this->registered_categories[ $slug ] = $category;
+		return $category;
+	}
+
+	/**
+	 * Unregisters a category.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_unregister_ability_category()` function.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @see wp_unregister_ability_category()
+	 *
+	 * @param string $slug The slug of the registered category.
+	 * @return ?\WP_Ability_Category The unregistered category instance on success, null on failure.
+	 */
+	public function unregister( string $slug ): ?WP_Ability_Category {
+		if ( ! $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				sprintf( esc_html__( 'Category "%s" not found.' ), esc_attr( $slug ) ),
+				'0.3.0'
+			);
+			return null;
+		}
+
+		$unregistered_category = $this->registered_categories[ $slug ];
+		unset( $this->registered_categories[ $slug ] );
+
+		return $unregistered_category;
+	}
+
+	/**
+	 * Retrieves the list of all registered categories.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_get_ability_categories()` function.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @see wp_get_ability_categories()
+	 *
+	 * @return array<string,\WP_Ability_Category> The array of registered categories.
+	 */
+	public function get_all_registered(): array {
```

Comment:
Same as the other comment. I think this is more intuitively true.
```suggestion
	public function get_all_categories(): array {
```

Reply:
By the way, the same names, `get_all_registered`, `get_registered`, `is_registered` are also used in the abilities registry. So we can change it in one go for both registries, but I would keep it outside of this PR so it doesn't block the work here.

Reply:
I chose `get_all_registered()` to maintain consistency with the naming pattern used in `WP_Abilities_Category_Registry`